### DISCUSSION TITLE
Allow leading zeros at the beginning of the number

### DISCUSSION THREAD

**Discussion by nghuuphuoc** [uNmAnNeR/imaskjs#734] [Q&A] [answered]
Here is how we use the [text-mask](https://github.com/text-mask/text-mask):

https://user-images.githubusercontent.com/4226036/189815058-b28271d9-dbe4-4c4a-9098-3ec362aa1d4e.mp4

Currently, it's not possible with the number masked. Do you have any guide or workaround, @uNmAnNeR?

**Comment by uNmAnNeR** [comment, marked as answer]
[yes](https://jsfiddle.net/uNmAjs/ayq9u631/)

**Reply by nghuuphuoc** [reply]
Thanks you @uNmAnNeR for the sample demo. However, there're some limitations/issues

* It use the `thousandsSeparator`, hence it doesn't format number as `$123,456,789` anymore
* It's not possible to enter zero at the beginning. I am only able to enter zero after I really enter a number, then jump to the beginning to enter zero there
* When the input loses the focus, its value removes all the zeros at the beginning.

**Reply by uNmAnNeR** [reply]
1. not sure i get the idea but you can try
```ts
     num: {
        mask: Number,
        thousandsSeparator: ',',
        radix: '.',
        mapToRadix: [],
      }
```
2. 3. true. but can you explain why do you need it? seems like usually numbers should not include zeros at the beginning.

**Reply by nghuuphuoc** [reply]
> but can you explain why do you need it? seems like usually numbers should not include zeros at the beginning

It is required by our customers for some reason.

**Reply by uNmAnNeR** [reply]
@nghuuphuoc to allow zeros at the beginning try to use `normalizeZeros: false`

**Comment by sebestenyb** [comment]
I have a similar headache, using the mask for a PIN code.

If I set the `mask: Number`, then unable to have codes starting with 0.
If I use a mask of `mask:'0000'`, then it works, but I can also enter `+`, `-`, and `.` signs. These eventually really mess up the input, allowing me to enter longer than 4-digit numbers.

Any idea to fix it?

PS: `normalizeZeros: false` doesn't seem to work, but I am only on `v6.4.3`
